lib/needle.js

```diff
@@ -432,16 +440,17 @@ Needle.prototype.send_request = function(count, method, uri, config, post_data,
           delete config.headers['content-length']; // in case the original was a multipart POST request.
         }
 
-        if (config.follow_set_cookies && resp.cookies)
-          config.headers['cookie'] = cookies.write(resp.cookies);
+        if (config.follow_set_cookies && resp.cookies) {
+          config.headers['Cookie'] = cookies.write(Object.assign(resp.cookies, cookies.read(headers['set-cookie'])));
+        }
 
         if (config.follow_set_referer)
           config.headers['referer'] = uri; // the original, not the destination URL.
 
         config.headers['host'] = null; // clear previous Host header to avoid conflicts.
 
         debug('Redirecting to ' + url.resolve(uri, headers.location));
-        return self.send_request(++count, method, url.resolve(uri, headers.location), config, post_data, out, callback);
+        return self.send_request(++count, method, url.resolve(uri, headers.location), config, post_data, out, cookies.read(headers['set-cookie']), callback);
```

Comment:
Instead of using cookies.read() again here, why not just use `resp.cookies` directly?